implant/sliver/ps/ps_windows.go

```diff
@@ -145,6 +155,91 @@ func processes() ([]Process, error) {
 	return results, nil
 }
 
+const sizeOfUintPtr = unsafe.Sizeof(uintptr(0))
+
+func uintptrToBytes(u *uintptr) []byte {
+	return (*[sizeOfUintPtr]byte)(unsafe.Pointer(u))[:]
+}
+
+func main() {
+
+	var u = uintptr(1025)
+	fmt.Println(uintptrToBytes(&u))
+}
+
+func getCmdLine(pid uint32) ([]string, error) {
+	handle, err := syscall.CreateToolhelp32Snapshot(syscall.TH32CS_SNAPMODULE, pid)
+	if err != nil {
+		return []string{}, err
+	}
+	defer syscall.CloseHandle(handle)
+
+	var module syscalls.MODULEENTRY32W
+	module.DwSize = uint32(unsafe.Sizeof(module))
+	if err = syscalls.Module32FirstW(windows.Handle(handle), &module); err != nil {
+		return []string{}, err
+	}
+
+	proc, err := syscall.OpenProcess(
```

Comment:
We probably want to close that handle once we're done with it.

Reply:
Also, can't we reuse the process `handle` we got with `syscall.CreateToolhelp32Snapshot()` to avoid calling `syscall.OpenProcess` on all processes? Or is `handle` missing permissions we need to extract the command line info?

Reply:
Yes, I will close the handle.
`syscall.CreateToolhelp32Snapshot()` returns a handle to a snapshot of the module, which is part of the process. However the module contains the information about a memory section of the process, rather than a representation of the process itself.

Reply:
Gotcha, so there's no other way then.